packages/component-library/src/components/form/mt-unit-field/mt-unit-field.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+  <div class="mt-unit-field">
+    <mt-number-field
+      :model-value="modelValue"
+      :placeholder="placeholder"
+      :number-type="numberType"
+      :step="step"
+      :min="min"
+      :max="max"
+      :digits="digits"
+      :fill-digits="fillDigits"
+      :allow-empty="allowEmpty"
+      :number-align-end="numberAlignEnd"
+      :disabled="disabled"
+      :required="required"
+      :is-inherited="isInherited"
+      :is-inheritance-field="isInheritanceField"
+      :disable-inheritance-toggle="disableInheritanceToggle"
+      :copyable="copyable"
+      :copyable-tooltip="copyableTooltip"
+      :help-text="helpText"
+      :name="name"
+      :size="size"
+      :label="label"
+      @update:model-value="$emit('update:modelValue', $event)"
+      @inheritance-restore="$emit('inheritance-restore')"
+      @inheritance-remove="$emit('inheritance-remove')"
+    >
+      <template #error>
+        <mt-field-error v-if="error" :error="error" />
+      </template>
+
+      <template #_unit-suffix>
+        <mt-unit-select
+          :model-value="defaultUnit"
+          :units="availableUnits"
+          :z-index="zIndex"
+          :disabled="disabled || isInherited"
+          @update:model-value="onUnitChange"
+        />
+      </template>
+    </mt-number-field>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed, watch } from "vue";
+import MtNumberField from "../mt-number-field/mt-number-field.vue";
+import MtFieldError from "../_internal/mt-field-error/mt-field-error.vue";
+import MtUnitSelect from "../_internal/mt-unit-select/mt-unit-select.vue";
+import convert from "convert-units";
+import type { Unit } from "convert-units";
```

Comment:
Maybe it makes more sense to just use a custom built union type with string literals so: "mm" | "cm" and so on.

That way we only allow to convert from and to units we make available.